### ISSUE TITLE
SnackBar / Toast message content delay

### ISSUE THREAD

**Issue by johnqiuwan** [angular/material2#1760]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
The Snackbar show up with message content , and the message content should NOT be delay 

#### What is the current behavior?
The SnackBar show up with empty content. The message content has a huge delay ( > 8 secs) to show up.

#### What are the steps to reproduce?

```ts
let user: any = { email: this.email, password: this.password };
  this.auth.createUser(user)
    .then((thisUser) => { 
      this.loading = false;
      this.errorMessage = null; 
      thisUser.auth.sendEmailVerification()
        .then(()=>{
          this.emailSentNotification();               
        })


emailSentNotification() {
  let message: string = "balh balh ..."
  let config: MdSnackBarConfig = new MdSnackBarConfig(this.viewContainerRef);
  this.snackBar.open(message, 'OK, Gotcha', config);
}
```

#### What is the use-case or motivation for changing an existing behavior?
Send email notification after user create account successful

#### Which versions of Angular, Material, OS, browsers are affected?
Angular2 / Angular2 Material(^2.0.0-alpha.9-3) / Windows 8 / Chrome

#### Is there anything else we should know?
No

**Comment by RoxKilly** [comment]
Please use three tick marks on the line above and the line below your code to keep code formatting. The tick mark is the character to the left of `1` on a US keyboard.

```
3 tick marks above this line
3 tick marks below this line
```

As for your issue, notice that the snackbar is shown only after all promises are resolved. The delay may be due to the time it takes for `createUser` + the time it takes for `sendEmailVerification` to complete.

To test this, replace the snackbar code with a vanilla Javascript alert and notice how long it takes for the alert to pop up. If it takes just as long, snackbar is not to blame.

Replace:
```
let config: MdSnackBarConfig = new MdSnackBarConfig(this.viewContainerRef);
this.snackBar.open(message, 'OK, Gotcha', config);
```
With:
```
alert(message);
```

**Comment by jelbourn** [comment]
CC @josephperrott 

@johnqiuwan Can you reproduce this issue in a Plunker? It sounds like it may be a situation when change detection isn't being run after the snackbar is loaded.

**Comment by josephperrott** [comment]
@johnqiuwan 
Let me konw if you are able to reproduce this.
I was unable to create a reproduction in a [plunkr](http://plnkr.co/edit/goraLjmDFTplWs7ypMq5?p=preview), even when explicitly resolving the promise outside of the angular zone.

**Comment by jelbourn** [comment]
We haven't been able to reproduce this issue. Feel free to open a new issue if you are able to provide a reproduction via Plunker.
